### ISSUE TITLE
Anyway to squeeze generated javascripts ?

### ISSUE THREAD

**Issue by cometta** [hpehl/piriti#13]
i use lot of modals and end up 66k of piriti json.read, json.write  compiled java scripts. Anyway to reduce output size?

**Comment by cometta** [comment]
reason need to reduce is because this slow down initial download of gwt if too many modals pojo.

**Comment by hpehl** [comment]
Yes I know this is a problem. Unfortunately I had no time to address this. I'm sure there's room for optimization, but currently I do not have time to take care of it. 

Anybody willing to jump in?

**Comment by cometta** [comment]
i still figuring how to optimize this. I think the rest of the community will be interested in this.

**Comment by cometta** [comment]
any suggestion on technique can be used ?

**Comment by lathil** [comment]
I wonder if Piriti is the right place for the optimisation. Doesn't it just generate *.Java files that GWT then compile into Javascript. Wouldn't it be better to the optimisation on GWT side ?

**Comment by hpehl** [comment]
Good point. But I think there's also a chance to reduce the generated java code. I'm not sure how much we could reduce the size, a first guess is something around 10%. But that's just a rough estimation. It's been a while since I worked on the codebase.
